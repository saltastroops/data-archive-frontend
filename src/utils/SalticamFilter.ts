export type SalticamFilter =
  | "SalticamFilter_Name"
  | "340-35"
  | "380-40"
  | "B-S1"
  | "CLR-S1"
  | "Halpha-S1"
  | "Hbn-S1"
  | "Hbw-S1"
  | "I-S1"
  | "R-S1"
  | "Sb-S1"
  | "SDSSg-S1"
  | "SDSSi-S1"
  | "SDSSr-S1"
  | "SDSSu-S1"
  | "SDSSz-S1"
  | "SR613-21"
  | "SR708-25"
  | "SR815-29"
  | "SR862-32"
  | "Su-S1"
  | "Sv-S1"
  | "Sy-S1"
  | "U-S1"
  | "V-S1";

export interface ISalticamFilter {
  name: SalticamFilter;
  descriptiveName: string;
}
