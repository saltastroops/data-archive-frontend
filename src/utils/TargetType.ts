export type TargetType = string;

export const GALAXY = "10";

export const ISM = "20";

export const SOLAR_SYSTEM_BODY = "30";

export const STAR = "40";
