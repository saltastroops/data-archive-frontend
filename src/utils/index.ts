export const toTitleCase = (s: string) => {
  s = s.replace(/_/g, " ");
  return s.replace(/\w\S*/g, (txt: string) => {
    return txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase();
  });
};
