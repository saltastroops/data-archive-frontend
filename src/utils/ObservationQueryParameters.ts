// TypeScript interfaces describing the state of the search form

import { SalticamFilter } from "./SalticamFilter";
import { TargetType } from "./TargetType";

/**
 * An interface describing the React state of the observation query form.
 *
 * This state is used for populating all search form fields (and error messages)
 * and is parsed into the GraphQL query for searching observations.
 *
 * Properties:
 * -----------
 * general:
 *     General information about the observation.
 * target:
 *     Target details.
 * telescope:
 *     Telescope (and instrument) details.
 */
// TODO: Remove loading, replace with IObservationQueryParameters
export interface ISearchFormState {
  general: IGeneral;
  target: ITarget;
  telescope?: ITelescope;
  loading: boolean;
}

/**
 * The available calibration types.
 */
export type CalibrationType = "arc" | "bias" | "flat" | "standard";

/**
 * An interface for query parameters related to general information.
 *
 * Properties:
 * -----------
 * errors:
 *     Error messages.
 * observationNight:
 *     Date of the night when the observation was taken. This
 *     refers to the date whe the night started.
 * proposalCode:
 *     Identifier of the proposal to which the observation belongs.
 * proposalTitle:
 *     Title of the proposal to which the observation belongs.
 * calibrations:
 *     Calibration types to include in the search results
 *
 */
export interface IGeneralErrors {
  observationNight?: string;
  principalInvestigator?: string;
  proposalCode?: string;
  proposalTitle?: string;
  calibrations?: string;
}
export interface IGeneral {
  errors: IGeneralErrors;
  observationNight?: string;
  principalInvestigator?: string;
  proposalCode?: string;
  proposalTitle?: string;
  calibrations: Set<CalibrationType>;
}

/**
 * The available target name resolvers.
 */
export type TargetResolver = "Simbad" | "NED" | "VizieR";

/**
 * The available units for the search cone radius.
 */
export type SearchConeRadiusUnits = "arcseconds" | "arcminutes" | "degrees";

/**
 * An interface for query parameters related to target details.
 *
 * Properties:
 * -----------
 * declination:
 *     Declination.
 * errors:
 *     Errors.
 * name:
 *     Target name.
 * resolver:
 *     Resolver to use for resolving the target name to a position.
 * rightAscension:
 *     Right ascension.
 * searchConeRadius:
 *     Radius of the search cone radius, in the units given by
 *     searchConeRadiusUnits.
 * searchConeRadiusUnits:
 *     Units of the search cone radius given by searchConeRadius.
 *
 */
export interface ITarget {
  declination?: string;
  errors: ITargetErrors;
  loading?: string;
  name?: string;
  resolver: TargetResolver;
  rightAscension?: string;
  searchConeRadius?: string;
  searchConeRadiusUnits: SearchConeRadiusUnits;
  targetTypes: Set<TargetType>;
}

export interface ITargetErrors {
  name?: string;
  declination?: string;
  resolver?: string;
  rightAscension?: string;
  searchConeRadius?: string;
  searchConeRadiusUnits?: string;
  targetTypes?: string;
}

// TELESCOPES

/**
 * The supported telescope names.
 */
export type TelescopeName = "Lesedi" | "SALT" | "1.9 m";

/**
 * An interface for query parameters related to a telescope.
 *
 * This serves as a placeholder for more concrete interfaces such as ISALT.
 *
 * Properties:
 * -----------
 * name:
 *     Telescope name.
 */
export interface ITelescope {
  name?: TelescopeName;
}

/**
 * An interface for query parameters related to SALT.
 *
 * Properties:
 * -----------
 * instrument:
 *     A SALT instrument.
 * name:
 *     The string "SALT".
 */
export interface ISALT extends ITelescope {
  instrument: IInstrument;
  name: "SALT";
}

/**
 * An interface for query parameters related to Lesedi.
 *
 * Properties:
 * -----------
 * instrument:
 *     An instrument used on Lesedi.
 * name:
 *     The string "Lesedi".
 */
export interface ILesedi extends ITelescope {
  instrument: IInstrument;
  name: "Lesedi";
}

/**
 * An interface for query parameters related to the 1.9 m Telescope.
 *
 * Properties:
 * -----------
 * instrument:
 *     An instrument used on the 1./9 m Telescope.
 * name:
 *     The string "1.9 m".
 */
export interface IOneDotNineM extends ITelescope {
  instrument: IInstrument;
  name: "1.9 m";
}

// INSTRUMENTS

/**
 * The available instrument names.
 */
export type InstrumentName =
  | "Salticam"
  | "RSS"
  | "HRS"
  | "BVIT"
  | "SpUpNIC"
  | "SHOC"
  | "HIPPO";

/**
 * An interface for query parameters related to an instrument.
 *
 * This serves as a placeholder for more concrete interfaces such as ISalticam.
 *
 * Properties:
 * -----------
 * name:
 *     Telescope name.
 */
export interface IInstrument {
  name: InstrumentName;
}

/**
 * An interface for query parameters related to Salticam.
 *
 * Properties:
 * -----------
 * detectorMode:
 *     Detector mode.
 * errors:
 *     Errors.
 * exposureTime:
 *     Exposure time.
 * filter:
 *     Filter.
 * name:
 *     The string "Salticam".
 */
export interface ISalticam extends IInstrument {
  detectorMode?: "Normal" | "Slot Mode";
  errors: {
    detectorMode?: string;
    exposureTime?: string;
    filter?: string;
  };
  exposureTime?: string;
  filter?: SalticamFilter;
  name: "Salticam";
}

/**
 * An interface for query parameters related to RSS.
 *
 * Properties:
 * -----------
 * detectorMode:
 *     Detector mode.
 * errors:
 *     Errors.
 * exposureTime:
 *     Exposure time.
 * filter:
 *     Filter.
 * mode:
 *     Instrument mode specific details.
 * name:
 *     The string "RSS".
 */
export interface IRSS extends IInstrument {
  detectorMode?: RSSDetectorMode;
  errors: {
    detectorMode?: string;
    exposureTime?: string;
    filter?: string;
  };
  exposureTime?: string;
  filter?: string;
  modes?: IRSSModes;
  name: "RSS";
}

export interface IRSSModes {
  errors: {
    fabryPerotMode?: string;
    grating?: string;
    polarimetryModes?: string;
  };
  fabryPerotMode?: RSSFabryPerotMode;
  grating?: RSSGrating;
  names: Set<RSSInstrumentMode>;
  polarimetryModes?: Set<RSSPolarimetryMode>;
}

export type RSSDetectorMode = "Normal" | "Slot Mode" | "";

export type RSSGrating =
  | "Open"
  | "pg0300"
  | "pg0900"
  | "pg1300"
  | "pg1800"
  | "pg2300"
  | "pg3000"
  | "";

export type RSSInstrumentMode =
  | "Fabry Perot"
  | "FP polarimetry"
  | "Imaging"
  | "Polarimetric imaging"
  | "MOS"
  | "MOS polarimetry"
  | "Spectropolarimetry"
  | "Spectroscopy";

export type RSSFabryPerotMode = "HR" | "LR" | "MR" | "TF" | "";

export type RSSPolarimetryMode =
  | "All Stokes"
  | "Circular"
  | "Linear"
  | "Linear Hi"
  | "";

/**
 * An interface for query parameters related to HRS.
 *
 * Properties:
 * -----------
 * errors:
 *     Errors.
 * exposureTime:
 *     Exposure time.
 * mode:
 *     Resolution mode.
 * name:
 *     The string "HRS".
 */
export interface IHRS extends IInstrument {
  errors: {
    exposureTime?: string;
    mode?: string;
  };
  exposureTime?: string;
  mode?: HRSMode;
  name: "HRS";
}

export type HRSMode =
  | "High Resolution"
  | "High Stability"
  | "Int Cal Fibre"
  | "Low Resolution"
  | "Medium Resolution";

export interface IBVIT extends IInstrument {
  errors: {};
  filter?: BVITFilter;
  mode?: "Imaging" | "Streaming";
  name: "BVIT";
}

export type BVITFilter = "B" | "H-alpha" | "Open" | "R" | "U" | "V";

/**
 * An interface for query parameters related to HIPPO.
 *
 * Properties:
 * -----------
 * errors:
 *     Errors.
 * exposureTime:
 *     Exposure time.
 * filter:
 *     Filter.
 * name:
 *     The string "HIPPO".
 */
export interface IHIPPO extends IInstrument {
  errors: {
    exposureTime?: string;
    filter?: string;
  };
  exposureTime?: string;
  filter?: string;
  name: "HIPPO";
}

/**
 * An interface for query parameters related to SHOC.
 *
 * Properties:
 * -----------
 * errors:
 *     Errors.
 * exposureTime:
 *     Exposure time.
 * filter:
 *     Filter.
 * name:
 *     The string "SHOC".
 */
export interface ISHOC extends IInstrument {
  errors: {
    exposureTime?: string;
    filter?: string;
  };
  exposureTime?: string;
  filter?: string;
  name: "SHOC";
}

/**
 * An interface for query parameters related to SpUpNIC.
 *
 * Properties:
 * -----------
 * errors:
 *     Errors.
 * exposureTime:
 *     Exposure time.
 * filter:
 *     Filter.
 * name:
 *     The string "SpUpNIC".
 */
export interface ISpUpNIC extends IInstrument {
  errors: {
    exposureTime?: string;
    filter?: string;
  };
  exposureTime?: string;
  filter?: string;
  name: "SpUpNIC";
}

export interface IFile {
  id: string;
  filename: string;
  name: string;
  dataType: string;
  isReduced: boolean;
  targetName: string;
  rightAscension: number;
  declination: number;
  observationNight: string;
  category: string;
  telescope: string;
  instrument: string;
  observationId?: string;
  url?: string;
}
export interface IObservation {
  id: string;
  name: string | "Unknown Observation";
  proposal?: string;
  telescope: string;
  startTime: string;
  files: IFile[];
}
