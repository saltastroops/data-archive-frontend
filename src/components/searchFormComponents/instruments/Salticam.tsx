import * as React from "react";
import { ISalticam } from "../../../utils/ObservationQueryParameters";
import { ISalticamFilter } from "../../../utils/SalticamFilter";
import { MainGrid, SubGrid } from "../../basicComponents/Grids";
import SelectField, { AnyOption } from "../../basicComponents/SelectField";

interface ISalticamProps {
  onChange: (value: any) => void;
  salticam: ISalticam;
}

/**
 * The list of Salticam filters which can be searched for.
 */
export const salticamFilters: ISalticamFilter[] = [
  { name: "U-S1", descriptiveName: "Johnson U" },
  { name: "B-S1", descriptiveName: "Johnson B" },
  { name: "V-S1", descriptiveName: "Johnson V" },
  { name: "R-S1", descriptiveName: "Cousins R" },
  { name: "I-S1", descriptiveName: "Cousins I" },
  { name: "380-40", descriptiveName: "380nm 40nm FWHM" },
  { name: "340-35", descriptiveName: "340nm 35nm FWHM" },
  { name: "CLR-S1", descriptiveName: "Fused silica clear" },
  { name: "SDSSu-S1", descriptiveName: "SDSS u'" },
  { name: "SDSSg-S1", descriptiveName: "SDSS g'" },
  { name: "SDSSr-S1", descriptiveName: "SDSS r'" },
  { name: "SDSSi-S1", descriptiveName: "SDSS i'" },
  { name: "SDSSz-S1", descriptiveName: "SDSS z'" },
  { name: "Su-S1", descriptiveName: "Stroemgren u" },
  { name: "Sb-S1", descriptiveName: "Stroemgren b" },
  { name: "Sv-S1", descriptiveName: "Stroemgren v" },
  { name: "Sy-S1", descriptiveName: "Stroemgren y" },
  { name: "Halpha-S1", descriptiveName: "H-alpha" },
  { name: "Hbw-S1", descriptiveName: "H-beta wide" },
  { name: "Hbn-S1", descriptiveName: "H-beta narrow" },
  { name: "SR613-21", descriptiveName: "SRE 1" },
  { name: "SR708-25", descriptiveName: "SRE 2" },
  { name: "SR815-29", descriptiveName: "SRE 3" },
  { name: "SR862-32", descriptiveName: "SRE 4" }
];

/**
 * A form for selecting Salticam-related search parameters.
 */
const Salticam = (props: ISalticamProps) => {
  const { onChange, salticam } = props;

  // Function for handling change events
  const change = (e: React.ChangeEvent<HTMLSelectElement>) => {
    const name = e.target.name;
    const value = e.target.value;
    onChange({
      ...salticam,
      [name]: value
    });
  };

  const detectorModes = ["Normal", "Slot Mode", "Drift Scan", "Frame Transfer"];

  return (
    <MainGrid>
      {/* Detector mode */}
      <SubGrid>
        <p>Detector Mode</p>
        <SelectField
          name="detectorMode"
          value={salticam.detectorMode}
          onChange={change}
        >
          <AnyOption />
          {detectorModes.map(detectorMode => (
            <option key={detectorMode} value={detectorMode}>
              {detectorMode}
            </option>
          ))}
        </SelectField>
      </SubGrid>

      {/* Filter */}
      <SubGrid>
        <p>Filter</p>
        <SelectField name={"filter"} value={salticam.filter} onChange={change}>
          <AnyOption />
          {salticamFilters.map(filter => (
            <option key={filter.name} value={filter.name}>
              {filter.descriptiveName}
            </option>
          ))}
        </SelectField>
      </SubGrid>
    </MainGrid>
  );
};

export default Salticam;
