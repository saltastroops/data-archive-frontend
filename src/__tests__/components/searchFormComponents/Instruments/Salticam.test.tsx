import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import * as React from "react";
import Salticam from "../../../../components/searchFormComponents/instruments/Salticam";
import { ISalticam } from "../../../../utils/ObservationQueryParameters";

describe("Salticam ", () => {
  it("should be defined", () => {
    const onChange = jest.fn();
    expect(
      mount(
        <Salticam
          salticam={{ errors: {}, name: "Salticam" }}
          onChange={onChange}
        />
      )
    ).toBeDefined();
  });

  it("should render correctly without detector mode and filter", () => {
    // Use mount instead of shallow for better snapshots
    const onChange = jest.fn();
    expect(
      toJson(
        mount(
          <Salticam
            salticam={{ errors: {}, name: "Salticam" }}
            onChange={onChange}
          />
        )
      )
    ).toMatchSnapshot();
  });

  it("should render correctly with detector mode and filter", () => {
    // Use mount instead of shallow for better snapshots
    const onChange = jest.fn();
    expect(
      toJson(
        mount(
          <Salticam
            salticam={{
              detectorMode: "Normal",
              errors: {},
              filter: "V-S1",
              name: "Salticam"
            }}
            onChange={onChange}
          />
        )
      )
    ).toMatchSnapshot();
  });

  it("should call the onChange method if the detector mode changes", () => {
    const onChange = jest.fn();
    const salticam: ISalticam = {
      detectorMode: "Normal",
      errors: {},
      name: "Salticam"
    };
    const wrapper = mount(<Salticam onChange={onChange} salticam={salticam} />);
    const detectorModeSelect = wrapper.find('[name="detectorMode"] select');
    detectorModeSelect.simulate("change", {
      target: { name: "detectorMode", value: "Slot Mode" }
    });
    expect(onChange).toHaveBeenCalledWith({
      ...salticam,
      detectorMode: "Slot Mode"
    });
  });

  it("should call the onChange method if the filter changes", () => {
    const onChange = jest.fn();
    const salticam: ISalticam = {
      errors: {},
      filter: "V-S1",
      name: "Salticam"
    };
    const wrapper = mount(<Salticam onChange={onChange} salticam={salticam} />);
    const detectorModeSelect = wrapper.find('[name="filter"] select');
    detectorModeSelect.simulate("change", {
      target: { name: "filter", value: "B-S1" }
    });
    expect(onChange).toHaveBeenCalledWith({ ...salticam, filter: "B-S1" });
  });
});
