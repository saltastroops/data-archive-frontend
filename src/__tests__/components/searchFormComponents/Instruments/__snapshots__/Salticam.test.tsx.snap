// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Salticam  should render correctly with detector mode and filter 1`] = `
.c0.c0 {
  display: grid;
  grid-template-columns: 50% 50%;
  padding: 10px;
}

.c1.c1 {
  text-align: left;
  padding-left: 10px;
}

<Salticam
  onChange={[MockFunction]}
  salticam={
    Object {
      "detectorMode": "Normal",
      "errors": Object {},
      "filter": "V-S1",
      "name": "Salticam",
    }
  }
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [
            Object {
              "className": "main-grid",
            },
          ],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bZQynM",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "
  && {
    display: grid;
    grid-template-columns: 50% 50%;
    padding: 10px;
  }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bZQynM",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="main-grid c0"
      >
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [
                  Object {
                    "className": "sub-grid",
                  },
                ],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-dnqmqq",
                  "isStatic": true,
                  "lastClassName": "c1",
                  "rules": Array [
                    "
  && {
    text-align: left;
    padding-left: 10px;
  }
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-dnqmqq",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sub-grid c1"
            >
              <p>
                Detector Mode
              </p>
              <SelectField
                name="detectorMode"
                onChange={[Function]}
                value="Normal"
              >
                <div
                  className="control select "
                >
                  <select
                    name="detectorMode"
                    onChange={[Function]}
                    value="Normal"
                  >
                    <AnyOption>
                      <option
                        key="any"
                        value=""
                      >
                        Any
                      </option>
                    </AnyOption>
                    <option
                      key="Normal"
                      value="Normal"
                    >
                      Normal
                    </option>
                    <option
                      key="Slot Mode"
                      value="Slot Mode"
                    >
                      Slot Mode
                    </option>
                    <option
                      key="Drift Scan"
                      value="Drift Scan"
                    >
                      Drift Scan
                    </option>
                    <option
                      key="Frame Transfer"
                      value="Frame Transfer"
                    >
                      Frame Transfer
                    </option>
                  </select>
                </div>
              </SelectField>
            </div>
          </StyledComponent>
        </styled.div>
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [
                  Object {
                    "className": "sub-grid",
                  },
                ],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-dnqmqq",
                  "isStatic": true,
                  "lastClassName": "c1",
                  "rules": Array [
                    "
  && {
    text-align: left;
    padding-left: 10px;
  }
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-dnqmqq",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sub-grid c1"
            >
              <p>
                Filter
              </p>
              <SelectField
                name="filter"
                onChange={[Function]}
                value="V-S1"
              >
                <div
                  className="control select "
                >
                  <select
                    name="filter"
                    onChange={[Function]}
                    value="V-S1"
                  >
                    <AnyOption>
                      <option
                        key="any"
                        value=""
                      >
                        Any
                      </option>
                    </AnyOption>
                    <option
                      key="U-S1"
                      value="U-S1"
                    >
                      Johnson U
                    </option>
                    <option
                      key="B-S1"
                      value="B-S1"
                    >
                      Johnson B
                    </option>
                    <option
                      key="V-S1"
                      value="V-S1"
                    >
                      Johnson V
                    </option>
                    <option
                      key="R-S1"
                      value="R-S1"
                    >
                      Cousins R
                    </option>
                    <option
                      key="I-S1"
                      value="I-S1"
                    >
                      Cousins I
                    </option>
                    <option
                      key="380-40"
                      value="380-40"
                    >
                      380nm 40nm FWHM
                    </option>
                    <option
                      key="340-35"
                      value="340-35"
                    >
                      340nm 35nm FWHM
                    </option>
                    <option
                      key="CLR-S1"
                      value="CLR-S1"
                    >
                      Fused silica clear
                    </option>
                    <option
                      key="SDSSu-S1"
                      value="SDSSu-S1"
                    >
                      SDSS u'
                    </option>
                    <option
                      key="SDSSg-S1"
                      value="SDSSg-S1"
                    >
                      SDSS g'
                    </option>
                    <option
                      key="SDSSr-S1"
                      value="SDSSr-S1"
                    >
                      SDSS r'
                    </option>
                    <option
                      key="SDSSi-S1"
                      value="SDSSi-S1"
                    >
                      SDSS i'
                    </option>
                    <option
                      key="SDSSz-S1"
                      value="SDSSz-S1"
                    >
                      SDSS z'
                    </option>
                    <option
                      key="Su-S1"
                      value="Su-S1"
                    >
                      Stroemgren u
                    </option>
                    <option
                      key="Sb-S1"
                      value="Sb-S1"
                    >
                      Stroemgren b
                    </option>
                    <option
                      key="Sv-S1"
                      value="Sv-S1"
                    >
                      Stroemgren v
                    </option>
                    <option
                      key="Sy-S1"
                      value="Sy-S1"
                    >
                      Stroemgren y
                    </option>
                    <option
                      key="Halpha-S1"
                      value="Halpha-S1"
                    >
                      H-alpha
                    </option>
                    <option
                      key="Hbw-S1"
                      value="Hbw-S1"
                    >
                      H-beta wide
                    </option>
                    <option
                      key="Hbn-S1"
                      value="Hbn-S1"
                    >
                      H-beta narrow
                    </option>
                    <option
                      key="SR613-21"
                      value="SR613-21"
                    >
                      SRE 1
                    </option>
                    <option
                      key="SR708-25"
                      value="SR708-25"
                    >
                      SRE 2
                    </option>
                    <option
                      key="SR815-29"
                      value="SR815-29"
                    >
                      SRE 3
                    </option>
                    <option
                      key="SR862-32"
                      value="SR862-32"
                    >
                      SRE 4
                    </option>
                  </select>
                </div>
              </SelectField>
            </div>
          </StyledComponent>
        </styled.div>
      </div>
    </StyledComponent>
  </styled.div>
</Salticam>
`;

exports[`Salticam  should render correctly without detector mode and filter 1`] = `
.c0.c0 {
  display: grid;
  grid-template-columns: 50% 50%;
  padding: 10px;
}

.c1.c1 {
  text-align: left;
  padding-left: 10px;
}

<Salticam
  onChange={[MockFunction]}
  salticam={
    Object {
      "errors": Object {},
      "name": "Salticam",
    }
  }
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [
            Object {
              "className": "main-grid",
            },
          ],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bZQynM",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "
  && {
    display: grid;
    grid-template-columns: 50% 50%;
    padding: 10px;
  }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bZQynM",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="main-grid c0"
      >
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [
                  Object {
                    "className": "sub-grid",
                  },
                ],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-dnqmqq",
                  "isStatic": true,
                  "lastClassName": "c1",
                  "rules": Array [
                    "
  && {
    text-align: left;
    padding-left: 10px;
  }
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-dnqmqq",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sub-grid c1"
            >
              <p>
                Detector Mode
              </p>
              <SelectField
                name="detectorMode"
                onChange={[Function]}
              >
                <div
                  className="control select "
                >
                  <select
                    name="detectorMode"
                    onChange={[Function]}
                  >
                    <AnyOption>
                      <option
                        key="any"
                        value=""
                      >
                        Any
                      </option>
                    </AnyOption>
                    <option
                      key="Normal"
                      value="Normal"
                    >
                      Normal
                    </option>
                    <option
                      key="Slot Mode"
                      value="Slot Mode"
                    >
                      Slot Mode
                    </option>
                    <option
                      key="Drift Scan"
                      value="Drift Scan"
                    >
                      Drift Scan
                    </option>
                    <option
                      key="Frame Transfer"
                      value="Frame Transfer"
                    >
                      Frame Transfer
                    </option>
                  </select>
                </div>
              </SelectField>
            </div>
          </StyledComponent>
        </styled.div>
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [
                  Object {
                    "className": "sub-grid",
                  },
                ],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-dnqmqq",
                  "isStatic": true,
                  "lastClassName": "c1",
                  "rules": Array [
                    "
  && {
    text-align: left;
    padding-left: 10px;
  }
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-dnqmqq",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sub-grid c1"
            >
              <p>
                Filter
              </p>
              <SelectField
                name="filter"
                onChange={[Function]}
              >
                <div
                  className="control select "
                >
                  <select
                    name="filter"
                    onChange={[Function]}
                  >
                    <AnyOption>
                      <option
                        key="any"
                        value=""
                      >
                        Any
                      </option>
                    </AnyOption>
                    <option
                      key="U-S1"
                      value="U-S1"
                    >
                      Johnson U
                    </option>
                    <option
                      key="B-S1"
                      value="B-S1"
                    >
                      Johnson B
                    </option>
                    <option
                      key="V-S1"
                      value="V-S1"
                    >
                      Johnson V
                    </option>
                    <option
                      key="R-S1"
                      value="R-S1"
                    >
                      Cousins R
                    </option>
                    <option
                      key="I-S1"
                      value="I-S1"
                    >
                      Cousins I
                    </option>
                    <option
                      key="380-40"
                      value="380-40"
                    >
                      380nm 40nm FWHM
                    </option>
                    <option
                      key="340-35"
                      value="340-35"
                    >
                      340nm 35nm FWHM
                    </option>
                    <option
                      key="CLR-S1"
                      value="CLR-S1"
                    >
                      Fused silica clear
                    </option>
                    <option
                      key="SDSSu-S1"
                      value="SDSSu-S1"
                    >
                      SDSS u'
                    </option>
                    <option
                      key="SDSSg-S1"
                      value="SDSSg-S1"
                    >
                      SDSS g'
                    </option>
                    <option
                      key="SDSSr-S1"
                      value="SDSSr-S1"
                    >
                      SDSS r'
                    </option>
                    <option
                      key="SDSSi-S1"
                      value="SDSSi-S1"
                    >
                      SDSS i'
                    </option>
                    <option
                      key="SDSSz-S1"
                      value="SDSSz-S1"
                    >
                      SDSS z'
                    </option>
                    <option
                      key="Su-S1"
                      value="Su-S1"
                    >
                      Stroemgren u
                    </option>
                    <option
                      key="Sb-S1"
                      value="Sb-S1"
                    >
                      Stroemgren b
                    </option>
                    <option
                      key="Sv-S1"
                      value="Sv-S1"
                    >
                      Stroemgren v
                    </option>
                    <option
                      key="Sy-S1"
                      value="Sy-S1"
                    >
                      Stroemgren y
                    </option>
                    <option
                      key="Halpha-S1"
                      value="Halpha-S1"
                    >
                      H-alpha
                    </option>
                    <option
                      key="Hbw-S1"
                      value="Hbw-S1"
                    >
                      H-beta wide
                    </option>
                    <option
                      key="Hbn-S1"
                      value="Hbn-S1"
                    >
                      H-beta narrow
                    </option>
                    <option
                      key="SR613-21"
                      value="SR613-21"
                    >
                      SRE 1
                    </option>
                    <option
                      key="SR708-25"
                      value="SR708-25"
                    >
                      SRE 2
                    </option>
                    <option
                      key="SR815-29"
                      value="SR815-29"
                    >
                      SRE 3
                    </option>
                    <option
                      key="SR862-32"
                      value="SR862-32"
                    >
                      SRE 4
                    </option>
                  </select>
                </div>
              </SelectField>
            </div>
          </StyledComponent>
        </styled.div>
      </div>
    </StyledComponent>
  </styled.div>
</Salticam>
`;
